discount-details-function-settings-block/src/DiscountFunctionSettings.liquid

```diff
@@ -201,14 +201,15 @@
   function useExtensionData() {
     const { applyMetafieldChange, i18n, data, resourcePicker, query } =
       useApi(TARGET);
+    const metafields = data?.metafields ? JSON.stringify(data.metafields) : '';
```

Comment:
It feels weirds to force the consumer to do that. This makes me wondering if there was a recent change that made `metafields` being mutated in place (same object reference, different contents), which is not forcing an update of the dependency (React checks dependencies with `Object.is`).

I'm not super familiar with how the data is exposed through the API (`launchOptions.context?.api`) but I'm wondering if we should look there?

For example, this could be relevant: https://github.com/shop/world/pull/121608/files#r2240572620